flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/RowDataWrapper.java

```diff
@@ -80,7 +81,7 @@ public int size() {
     throw new UnsupportedOperationException("Could not set a field in the RowDataWrapper because rowData is read-only");
   }
 
-  private interface PositionalGetter<T> {
+  private interface PositionalGetter<T> extends Serializable {
```

Comment:
If these are going to extend serializable, do we need `serialVersionUID` somewhere?

Specifically for Flink, if any of implementations of this interface get persisted as state, will users have to lose state if they remove operators from their job graph? If they're not part of state, I suppose it's not mandatory for implementing classes

Reply:
Yeah, that seems reasonable to me. I would probably update the implementation classes.

We should also add tests to validate that the `PositionalGetter` implementations are serializable with Java and Kryo.